src/Prometheus/Internal/Pure/Summary.hs

```diff
@@ -109,27 +118,26 @@ insert value summaryData@(Estimator oldCount oldSum quantiles items) =
             -- the first item in a multi-item list. For subsequent steps of
             -- a multi valued list, this case cannot happen as it would have
             -- fallen through to the case below in the previous step.
-            | value <= _iValue x = Item value 1 0 : x : y : xs
-            | value <= _iValue y = x : Item value 1 (calcD $ r + _iG x)
-                                     : y : xs
-            | otherwise          = x : insertItem (_iG x + r) (y : xs)
+            | value <= iValue x = Item value 1 0 : x : y : xs
+            | value <= iValue y = x : Item value 1 (calcD $ r + iG x)
+                                    : y : xs
+            | otherwise         = x : insertItem (iG x + r) (y : xs)
 
         calcD r = max 0
                 $ floor (invariant summaryData (fromIntegral r)) - 1
 
 
 compress :: Estimator -> Estimator
 compress est@(Estimator _ _ _ items)    =
```

Comment:
Is this for saving space? If the answer is yes, why we are not using some other Data Structure like Piority Queue which implements LRU for free https://hackage.haskell.org/package/psqueues

Reply:
This algorithm is a blind copy from one of the other `prometheus` libraries. It obviously does some compression internally to keep everything space-consistent, but I don't know the details. (I've ran a test locally and 10 thousand observations compress to about 20 items in the `Estimator`)